u19_pipeline/recording.py

```diff
@@ -0,0 +1,181 @@
+    preprocessing_params: longblob  # dictionary of all applicable parameters
+    """
+
+    @classmethod
+    def insert_new_params(cls, recording_modality: str, paramset_idx: int,
+                          paramset_desc: str, params: dict):
+        param_dict = {'recording_modality': recording_modality,
+                      'preprocess_paramset_idx': paramset_idx,
+                      'preprocess_paramset_desc': paramset_desc,
+                      'preprocessing_params': params,
+                      'preprocess_param_set_hash':  dict_to_uuid(params)}
+        param_query = cls & {'preprocess_param_set_hash': param_dict['preprocess_param_set_hash']}
+
+        if param_query:  # If the specified param-set already exists
+            existing_paramset_idx = param_query.fetch1('preprocess_paramset_idx')
+            if existing_paramset_idx == paramset_idx:  # If the existing set has the same paramset_idx: job done
+                return
+            else:  # If not same name: human error, trying to add the same paramset with different name
+                raise dj.DataJointError(
+                    'The specified param-set'
+                    ' already exists - paramset_idx: {}'.format(existing_paramset_idx))
+        else:
+            cls.insert1(param_dict)
+
+
+@schema
+class ProcessingParamSet(dj.Lookup):
+    definition = """
+    # Parameter set to be used in the preprocessing steps
+    processing_paramset_idx:  smallint
+    ---
+    -> RecordingModality    
+    processing_paramset_desc: varchar(128)
+    processing_param_set_hash: uuid
+    unique index (processing_param_set_hash)
+    processing_params: longblob  # dictionary of all applicable parameters
+    """
+
+    @classmethod
+    def insert_new_params(cls, recording_modality: str, paramset_idx: int,
+                          paramset_desc: str, params: dict):
+        param_dict = {'recording_modality': recording_modality,
+                      'processing_paramset_idx': paramset_idx,
+                      'processing_paramset_desc': paramset_desc,
+                      'processing_params': params,
+                      'processing_param_set_hash':  dict_to_uuid(params)}
+        param_query = cls & {'processing_param_set_hash': param_dict['processing_param_set_hash']}
+
+        if param_query:  # If the specified param-set already exists
+            existing_paramset_idx = param_query.fetch1('processing_paramset_idx')
+            if existing_paramset_idx == paramset_idx:  # If the existing set has the same paramset_idx: job done
+                return
+            else:  # If not same name: human error, trying to add the same paramset with different name
+                raise dj.DataJointError(
+                    'The specified param-set'
+                    ' already exists - paramset_idx: {}'.format(existing_paramset_idx))
+        else:
+            cls.insert1(param_dict)
+
+@schema
+class Recording(dj.Manual):
```

Comment:
This would not effect the functionality of this module, but perhaps the `Recording` table should be placed directly after the `RecordingModality` table?

Reply:
All right,
My "logic" was to put all lookup tables first. 
But your proposal makes a lot of sense

Reply:
Thanks!  Either way works.  I am just used to ordering the tables in approximately the same order as how the data flows.